modules/basicfuncs/numeric-funcs.c

```diff
@@ -56,83 +111,204 @@ tf_num_parse(gint argc, GString *argv[],
 static void
 tf_num_plus(LogMessage *msg, gint argc, GString *argv[], GString *result)
 {
-  gint64 n, m;
+  struct value n, m;
 
   if (!tf_num_parse(argc, argv, "+", &n, &m))
     {
       g_string_append_len(result, "NaN", 3);
       return;
     }
 
-  format_int64_padded(result, 0, ' ', 10, n + m);
+  if (n.value_type == VALUE_TYPE_GINT64 && m.value_type == VALUE_TYPE_GINT64)
+    {
+      format_int64_padded(result, 0, ' ', 10, value_as_int(n) + value_as_int(m));
+    }
+  else
+    {
+      gdouble res = value_as_double(n) + value_as_double(m);
+      g_string_append_printf(result, "%f", res);
+    }
```

Comment:
Please extract this part into a function (as used in many places) like `format_number`.